### ISSUE TITLE
Lags when drawing on 3D model

### ISSUE THREAD

**Issue by DearFox** [JannisX11/blockbench#2104]
### What are you trying to do, and what do you expect to happen?

When drawing a texture on 3D model, the program freezes for some time.
 The more polygons you have and the size of the texture, the stronger this will be felt. 
 In my case:
 I have a model - 
 ![image](https://github.com/JannisX11/blockbench/assets/59219907/48a138ec-09f8-4366-b1ee-6f08c9a0ee54)
 And the texture is 256x256 pixels
 When I try to draw something on a model in drawing mode, the program freezes for a few seconds, after which I get a line that supposedly runs from the start to the end point of drawing (ignoring bends and all that stuff)

Demonstration of this problem: 
![srw6uidtul](https://github.com/JannisX11/blockbench/assets/59219907/6301281c-7011-4aa6-bfd7-a9e686296abf)

### What happens instead?

The program lags, making it almost impossible to draw on the model.

### Model format in which the issue occurs

Generic Model

### Blockbench variant

Program

### Blockbench version

4.8.3

### Operating System

Windows 10 x64

### Installed Blockbench plugins

I checked this error in the web version without plugins, the same thing happens.

**Comment by DearFox** [comment]
How to recover the problem:
 1) Create a new model "Generic Model"
 2) Create a flat mesh
 3) Cut the mesh 10-20 times in two planes
 4) Create a texture 256x256 or larger, just a solid color texture
 5) Apply the texture to the previously cut plane 
 6) Go to drawing mode and draw on the model
 7) Enjoy broken line or freezes
 
![sreu5dtkj](https://github.com/JannisX11/blockbench/assets/59219907/eb63cad5-1ad1-4c95-8421-5556c1e8878e)

**Comment by DearFox** [comment]
I decided to test this fix, and although the model now still rotates when drawing on it, the amount of lag and stuttering has decreased.
 But I noticed something strange.  Now lags and freezes appear on different parts of the model to varying degrees.
 Here are examples when drawing using the same model:
![sret6udrt7iy_0](https://github.com/JannisX11/blockbench/assets/59219907/469e25cd-aac5-4590-a830-d0b792f6140b)
![sret6udrt7iy_1](https://github.com/JannisX11/blockbench/assets/59219907/eac9e24a-0b9f-41d9-a813-f11c60c91dbc)

According to the second example, it seems to me that freezes/lags appear when moving from one polygon to another, while when I drew a continuous line for a long time in this place, it seems that the transitions became smoother and the freezes were not felt so much.  But after I started a new line, it seems the lag have resumed.

Version Blockbench : 4.9.0 Beta 1 
Build and run from tree: https://github.com/JannisX11/blockbench/tree/4665122d6949fbb042a574b06156ca175f3c8923
